src/Identity.Clients/Services/ResourceService.cs

```diff
@@ -164,6 +172,72 @@ private void UpdateManagers(Data.Resource entity, IEnumerable<ResourceManager> m
             }
         }
 
+        private void UpdateSecrets(Data.Resource entity, IEnumerable<ApiSecret> secrets)
+        {
+            foreach (var secret in secrets.Where(s => s.Deleted))
+            {
+                var target = entity.Secrets.SingleOrDefault(s => s.Id == secret.Id);
+                if (target != null)
+                    entity.Secrets.Remove(target);
+            }
+        }
+
+        private void ValidateScopes(Data.Resource entity, string scopes, string previousName)
+        {
+            if (entity.Scopes == scopes && entity.Name == previousName)
+                return;
+            if (String.IsNullOrEmpty(scopes))
+                entity.Scopes = entity.Name;
+
+            var scopeNames = scopes.Split(" ", StringSplitOptions.RemoveEmptyEntries)
+                .Select(s => 
+                {
+                    if (entity.Name != previousName) // find and replace first occurance of previousName
+                    {
+                        var i = s.IndexOf(previousName); 
+                        string name = i < 0 ? s : s.Substring(0, i) + entity.Name + s.Substring(i + previousName.Length);
+                        return name == entity.Name || name.StartsWith($"{entity.Name}-") ? name : $"{entity.Name}-{name}";
+                    }
+                    if (s == entity.Name || (s.StartsWith($"{entity.Name}-")))
+                        return s;
+                    return $"{entity.Name}-{s}";
```

Comment:
This is somewhat complicated logic to, in the event of the resource name being changes, replace the old prefix of each scope name in the Scopes field with the new resource name, if they haven't manually done so already themselves. 

A `str.Replace(old_name, new_name)` for the whole string is clean, but would replace all occurrences of that substring regardless of it being the prefix. For example a resource called `dev` might have scope `dev-development`, changing the resource name to `dev1` would change the scope to `dev1-dev1elopment`. We could do `str.Replace(old_name+"-", new_name="-")` and that would mostly prevent this.

Not sure if this is a concern, or it we need to even change the scope prefix at all or just leave what is was before.